presc/misclassifications/misclass_rate.py

```diff
@@ -78,7 +106,36 @@ def misclass_rate_feature(
             test_dataset_misclass[feature], bins
         )
 
+        # compute metric
+        metric_list = []
+        # get all groups and compute their metrics for each group
+        for key in groups.groups.keys():
+            try:
+                theGroup = groups.get_group(key)
+                test_y = pd.Series(theGroup[y_test_name]).to_numpy()
+                pred_y = []
+                for row in theGroup.index:
+                    pred_y.append(y_pred_series.at[row])
+                sc = metric_function(test_y, pred_y)
+                metric_list.append(sc)
+            except KeyError:
+                metric_list.append(0)
+
     else:
+        # compute metric
+        groups = test_dataset.groupby(by=feature)
+
+        metric_list = []
+        # get all groups and compute their metrics for each group
+        for key in groups.groups.keys():
+            theGroup = groups.get_group(key)
+            test_y = pd.Series(theGroup[y_test_name]).to_numpy()
+            pred_y = []
+            for row in theGroup.index:
+                pred_y.append(y_pred_series.at[row])
+            sc = accuracy_score(test_y, pred_y)
+            metric_list.append(sc)
+
         # Histogram of all points for categorical features
         total_histogram_counts = test_dataset[feature].value_counts().sort_index()
```

Comment:
Similarly as  above, you can now move the rest of this outside of this function.

Reply:
I kept `total_histogram_counts = test_dataset[feature].value_counts().sort_index()`, since I want it to help me calculate bins after that:` bins = np.asarray(total_histogram_counts.index)`. This is the easiest way I can think to calculate bins for categorical data. Can you give me some advice about that?

Reply:
Since the bins are already given by the `groups` you could just use `list(groups.groups.keys())`.